apps/common/controllers/postcode-loop.js

```diff
@@ -50,7 +33,8 @@ module.exports = class PostcodeLoopController extends BaseController {
 
   saveValues(req, res, cb) {
     const saveValues = super.saveValues(req, res, cb);
-    this.postcodeChanged = req.form.values[`${this.field}-postcode`] !== req.sessionModel.get(`${this.field}-postcode`);
+    this.postcodeChanged = req.form.values[`${this.options.locals.field}-postcode`]
```

Comment:
@sulthan-ahmed To fix this, look for where the value of `this.postcodeChanged` is used, then swap `this.postcodeChanged` for `req.sessionModel.get('postcode-changed')` and `req.sessionModel.set('postcodechanged', ...)` here.

Then the `postcodeChanged` property is specific to the current user, and not shared across all implementations of the controller.